engine/src/execution_graph/logic_controllers/BatchJoinProcessing.h

```diff
@@ -146,11 +106,12 @@ class PartwiseJoin : public kernel {
 
 	std::unique_ptr<ral::frame::BlazingTable> load_right_set(){
 		this->max_right_ind++;
-		// need to load all the right side, if doing any type of outer join
-		bool load_all = this->join_type != INNER_JOIN ?  true : false;
-		auto table = load_set(this->right_sequence, this->input_.get_cache("input_b"), load_all);
+		std::unique_ptr<ral::frame::BlazingTable> table;
+		if (this->right_sequence.wait_for_next()){
+			table = this->right_sequence.next();
+		}
 		if (not right_schema && table != nullptr) {
-			right_schema = std::make_unique<TableSchema>(table->get_schema(),  table->names());
+			left_schema = std::make_unique<TableSchema>(table->get_schema(),  table->names());
```

Comment:
this should be right_schema

Reply:
OH SHIT good catch! I swear I double checked this. Triple checking is better